File: engine/engine-model/src/main/scala/tools/packet/ui/QuickSlotKey.scala
package tools.packet.ui

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class QuickSlotKey(private var _keyMap: Array[Byte]) extends PacketInput {
  def keyMap: Array[Byte] = _keyMap

  override def opcode(): SendOpcode = SendOpcode.QUICKSLOT_INIT
}

File: engine/engine-model/src/main/java/tools/packet/ui/QuickSlotKey.java
package tools.packet.ui;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record QuickSlotKey(byte[] keyMap) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.QUICK_SLOT_INIT;
   }
}